File: 백준/Silver/9996. 한국이 그리울 땐 서버에 접속하지/한국이 그리울 땐 서버에 접속하지.cc
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int N;
    string target, temp, front, back;

    cin >> N >> target;
    size_t pos = target.find('*');
    front = target.substr(0, pos);
    back = target.substr(pos + 1);

    for (int i = 0; i < N; i++)
    {
        cin >> temp;
        if (temp.size() < front.size() + back.size())
        {
            cout << "NE" << "\n";
            continue;
        }
        if (temp.substr(0, front.size()) != front || temp.substr(temp.size() - back.size()) != back)
        {
            cout << "NE" << "\n";
        }
        else {
            cout << "DA" << "\n";
        }
    }

    return 0;
}

File: 백준/Silver/9996. 한국이 그리울 땐 서버에 접속하지/한국이 그리울 땐 서버에 접속하지.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        String target = br.readLine();
        String front, back, temp;
        int pos;

        pos = target.indexOf("*");
        front = target.substring(0, pos);
        back = target.substring(pos + 1);

        for (int i = 0; i < N; i++) {
            temp = br.readLine();
            if (temp.length() < front.length() + back.length()) {
                System.out.println("NE");
                continue;
            }
            if (temp.substring(0, front.length()).equals(front)
                    && temp.substring(temp.length() - back.length()).equals(back)) {
                System.out.println("DA");
            } else {
                System.out.println("NE");
            }

        }
    }
}